webhook/README.md

```diff
@@ -0,0 +1,101 @@
+## Knative Webhooks
+
+Knative provides infrastructure for authoring webhooks under
+`knative.dev/pkg/webhook` and has a few built-in helpers for certain
+common admission control scenarios.  The built-in admission controllers
+are:
+1. Resource validation and defaulting (builds around `apis.Validatable`
+  and `apis.Defaultable` under `knative.dev/pkg/apis`).
+2. ConfigMap validation, which builds around similar patterns from
+  `knative.dev/pkg/configmap` (in particular the `store` concept)
+
+To illustrate standing up the webhook, let's start with one of these
+built-in admission controllers and then talk about how you can write
+your own admission controller.
+
+## Standing up a Webhook from an Admission Controller
+
+We provide facilities in `knative.dev/pkg/injection/sharedmain` to try and
+eliminate much of the boilerplate involved in standing up a webhook. For this
+example we will show how to stand up the webhook using the built-in admission
+controller for validating and defaulting resources.
+
+The code to stand up such a webhook looks roughly like this:
+
+```
+// Create a function matching this signature to pass into sharedmain.
+func NewResourceAdmissionController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+	return resourcesemantics.NewAdmissionController(ctx,
+		// Name of the resource webhook (created via yaml)
+		fmt.Sprintf("resources.webhook.%s.knative.dev", system.Namespace()),
+
+		// The path on which to serve the webhook.
+		"/resource-validation",
+
+		// The resources to validate and default.
+		map[schema.GroupVersionKind]resourcesemantics.GenericCRD{
+			// List the types to validate, this from knative.dev/sample-controller
+			v1alpha1.SchemeGroupVersion.WithKind("AddressableService"): &v1alpha1.AddressableService{},
+		},
+
+		// A function that infuses the context passed to Validate/SetDefaults with custom metadata.
+		func(ctx context.Context) context.Context {
+			// Here is where you would infuse the context with state
+			// (e.g. attach a store with configmap data, like knative.dev/serving attaches config-defaults)
+			return ctx
+		},
+
+		// Whether to disallow unknown fields when parsing the resources' JSON.
+		true,
+	)
+}
+
+func main() {
+	// Set up a signal context with our webhook options.
+	ctx := webhook.WithOptions(signals.NewContext(), webhook.Options{
+        // The name of the Kubernetes service selecting over this deployment's pods.
+		ServiceName: "webhook",
```

Comment:
Formatting seems off here and below.
Also same naming here? Would help to eradicate the comment.

Reply:
Fixed, tabs vs. spaces...